src/combat/combatant.js

```diff
@@ -1,3 +1,53 @@
+import { getCanvas } from '../utils/client-hooks';
 export default class YearZeroCombatant extends Combatant {
-  // TODO https://gitlab.com/peginc/swade/-/blob/develop/src/module/documents/SwadeCombatant.ts
-}
\ No newline at end of file
+
+  async setCardValue(cardValue) {
+    return this.setFlag('yzec', 'cardValue', cardValue);
+  }
+
+  get cardValue() {
+    return this.getFlag('yzce', 'cardValue');
+  }
+
+  get cardString() {
+    return this.getFlag('yzce', 'cardString');
+  }
+
+  async setCardString(cardString) {
+    return this.setFlag('yzce', 'cardString', cardString);
+  }
+  /**
+   *
+   * @param {Combatant} target    the combatant with which this combatant will swap a card
+   */
+  // TODO  recalculate the turn order on complettion
+  async swapInitiativeCard(target) {
+    const combatantCard = this.cardValue();
+    this.setCardValue(target.cardValue());
+    target.setCardValue(combatantCard);
+  }
+
+  /**
+   * @override
+   * @param {CombatantDataConstructorData} data
+   * @param {DocumentModificationOptions} options
+   * @param {User} user
+   */
+  async _preCreate(data, options, user) {
+    await super._preCreate(data, options, user);
+    const combatants = game?.combat?.combatants.size ?? 0;
+    const tokenID = data.tokenId instanceof TokenDocument ? data.tokenId.id : data.tokenId;
+    const tokenIndex =
+      getCanvas()
+        .tokens?.controlled.map(t => t.id)
+        .indexOf(tokenID) ?? 0;
+    const sortValue = tokenIndex + combatants;
+    this.data.update({
+      flags: {
+        yzce: {
```

Comment:
Not sure this is a valid flag scope if used with `setFlag` seeing as the module name is `yze-combat`

Reply:
not 100% sure but it can't hurt to set it to match just to be on the safe side :)

Reply:
It'll work. In my `blade-runner` game system, I use `bladerunner` for the flags.

Still I suppose it's best practice to put the same name to avoid future conflicts.